src/main/java/org/jabref/gui/EntryTypeViewModel.java

```diff
@@ -161,7 +161,7 @@ public void runFetcherWorker() {
                 } else {
                     // Regenerate CiteKey of imported BibEntry
                     new BibtexKeyGenerator(basePanel.getBibDatabaseContext(), prefs.getBibtexKeyPatternPreferences()).generateAndSetKey(entry);
-                    basePanel.insertEntry(entry);
+                    basePanel.insertEntry(entry); // Should this be basePanel.getDatabase().insertEntry(entry);?
```

Comment:
Okay, let me make sure I understand. It sounds like the `BasePanel.insertEntry` method is called when the user's action's primary purpose is to insert an entry, for example when the "insert entry" button is pressed. However, the `BibDatabase.insertEntry` method is called when the entry is inserted in the process of doing something else, for example when importing PDF content, and you wouldn't want an undo just for the entry insertion alone. Is that right?

Reply:
@tobiasdiez Is this the idea?

Reply:
I think `basePanel.insertEntry` is indeed the better choice, because of the extra work it does. `BibDatabase.insertEntry` is just the plain insertion of the entry in the database.